app/views/internal/users/_notes.erb

```diff
@@ -1,7 +1,7 @@
 <div class="row">
   <div class="col-12">
     <h2>Notes</h2>
-    <% if @user.notes.count > 0 %>
+    <% if @user.notes.load.size > 0 %>
```

Comment:
since we know that the notes are going to be iterated upon in the following line, we can preload them, to avoid a second `SELECT count` that would instead happen with `.each`

Reply:
Since it seems this partial will get rendered unconditionally, we could also `.includes(:notes)` on the query in the controller, which is something many people will be more familiar with. It also keeps AR calls out of the view, which IMHO is a good thing.

Reply:
I would keep this pattern for now, which I documented, so that it can be learned from the example. `.count` vs `.size` can be a tricky decision, leaving good patterns around seems a good idea :D Since there's no N + 1 anyway (we don't iterate on users) I would leave it as is for now

Reply:
@rhymes That's the thing, I'm not sure `.load.size` in a view is such a good pattern. 

* `.count`: result of SQL count
    ```ruby
     # SQL count
     u.identities.count
   (1.2ms)  SELECT COUNT(*) FROM "identities" WHERE "identities"."user_id" = $1  [["user_id", 2]]
    #=> 1
    ```
* `.length`: loads association into memory, then returns length of collection
    ```ruby
   # loads association, keeps it in memory
    u.identities.length
    Identity Load (1.5ms)  SELECT "identities".* FROM "identities" WHERE "identities"."user_id" = $1  [["user_id", 2]]
    #=> 1
    # it's been previously loaded, no more query
     u.identities.first
     #=> #<Identity:0x0000000117837530> { ... }
    ```
* `.size`: behaves like either of the other two, depending on whether the association has already been loaded or not. So `load.size` is basically `length`.
    ```ruby
    # Same query performed by `length` to load the association
    u.identities.load.size
    Identity Load (0.8ms)  SELECT "identities".* FROM "identities" WHERE "identities"."user_id" = $1  [["user_id", 2]]
    #=> 1
    # previously loaded, so no more query
    u.identities.first
    #<Identity:0x0000000117f78498> { ... }
    ```

Reply:
In the past I often told people to ignore `length` all together. If you care about not loading, use `count`. If you don't, use `size`. If you rely on `length` loading the association for you, something IMHO went wrong, since you could already have used `includes`, `preload` or similar.